hudi-hadoop-mr/src/main/java/org/apache/hudi/hadoop/utils/HoodieRealtimeInputFormatUtils.java

```diff
@@ -206,16 +206,28 @@ private static Configuration addProjectionField(Configuration conf, String field
 
   public static void addRequiredProjectionFields(Configuration configuration) {
     // Need this to do merge records in HoodieRealtimeRecordReader
-    addProjectionField(configuration, HoodieRecord.RECORD_KEY_METADATA_FIELD, HoodieInputFormatUtils.HOODIE_RECORD_KEY_COL_POS);
-    addProjectionField(configuration, HoodieRecord.COMMIT_TIME_METADATA_FIELD, HoodieInputFormatUtils.HOODIE_COMMIT_TIME_COL_POS);
-    addProjectionField(configuration, HoodieRecord.PARTITION_PATH_METADATA_FIELD, HoodieInputFormatUtils.HOODIE_PARTITION_PATH_COL_POS);
+    if (configuration.get(HoodieInputFormatUtils.HOODIE_USE_META_FIELDS) == null || configuration.get(HoodieInputFormatUtils.HOODIE_USE_META_FIELDS)
```

Comment:
so why do we ned this if we are setting this already at the record reader level?